src/offset_of.rs

```diff
@@ -145,4 +170,17 @@ mod tests {
 
         assert_eq!(offset_of!(sub::Foo, x), 0);
     }
+
+    #[cfg(feature = "unstable_const")]
+    #[test]
+    fn const_offset() {
+        #[repr(C)]
+        struct Foo {
+            a: u32,
+            b: [u8; 2],
+            c: i64,
+        }
+
+        assert_eq!([0; offset_of!(Foo, b)].len(), 4);
+    }
```

Comment:
Please add another test that actually exhibits the previous problem with generic functions.